### PR TITLE
Fix CustomObject bug destroy & Ragdoll Size & new effect & Decay AHP

### PR DESCRIPTION
**Pull request by warquys**

### CODE REVIEW

**Comment by GrafDimenzio** [comment]
GetVoltage() - Remove this it looks to me like HeavyController.ActiveGenerators * 1000 and more complicated

`ragdoll.netIdentity.UpdatePositionRotationScale()` - This basically sends a new Spawn message and since you spawn it literally the next line of Code you don't need it, also fix the formatting

`public static Synapse.Api.Ragdoll GetRagdoll(this Ragdoll rag) => Map.Get.Ragdolls.FirstOrDefault(x => x.ragdoll == null && x.ragdoll == rag);` I think this should be ragdoll != null

```
public static Door GetDoor(this Interactables.Interobjects.DoorUtils.DoorVariant door)
{
	return Map.Get.Doors.FirstOrDefault(x => x.VDoor != null && x.GameObject == door.gameObject);
}```
 It's not wrong but the formatting is weird compared to all others

File: Synapse/Api/Map.cs [hunk 1]
```diff
@@ -129,6 +129,7 @@ public void SendBroadcast(ushort time, string message, bool instant = false)
             ScpRecontainmentType.Chaos => $". SCP {scp} SUCCESSFULLY TERMINATED . BY CHAOSINSURGENCY",
             ScpRecontainmentType.Scientist => $". SCP {scp} SUCCESSFULLY TERMINATED . BY SCIENCE PERSONNEL",
             ScpRecontainmentType.ClassD => $". SCP {scp} SUCCESSFULLY TERMINATED . BY CLASSD PERSONNEL",
+            ScpRecontainmentType.Scp => $"TERMINATED BY SCP {Unit}",
```
Comment (AlmightyLks) [inline Synapse/Api/Map.cs:132]: This sounds like a direct communication to one specific player

However, this is for announcements across all players. Doesn't make much sense, nor do I understand the reasoning why you avoid the vanilla "Unknown"

**Comment by AlmightyLks** [comment]
Is there a reason why you added so many null checks for synapse object lists?

File: Synapse/SynapseExtensions.cs [hunk 2]
```diff
@@ -55,23 +55,24 @@ public static class SynapseExtensions
...
     public static Team GetTeam(this RoleType role) => Server.Get.Host.ClassManager.Classes.SafeGet(role).team;
 
-    public static Room GetSynapseRoom(this RoomIdentifier identifier) => Map.Get.Rooms.FirstOrDefault(x => x.Identifier == identifier);
+    public static Room GetSynapseRoom(this RoomIdentifier identifier) => Map.Get.Rooms.FirstOrDefault(x => x?.Identifier == identifier);
 
-    public static Generator GetGenerator(this Scp079Generator generator079) => Map.Get.Generators.FirstOrDefault(x => x.GameObject == generator079.gameObject);
+    public static Generator GetGenerator(this Scp079Generator generator079) => Map.Get.Generators.FirstOrDefault(x => x?.GameObject == generator079.gameObject);
 
-    public static Door GetDoor(this Interactables.Interobjects.DoorUtils.DoorVariant door) => Map.Get.Doors.FirstOrDefault(x => x.GameObject == door.gameObject);
+    public static Door GetDoor(this Interactables.Interobjects.DoorUtils.DoorVariant door) => Map.Get.Doors.FirstOrDefault(x => x.VDoor != null && x.GameObject == door.gameObject);
```
[the remark this answers is not in the document]
Comment (warquys) [reply Synapse/SynapseExtensions.cs:62]: the VDoor can be null if you destroy using unity one and dont remov it

File: Synapse/SynapseExtensions.cs [hunk 3]
```diff
@@ -55,23 +55,24 @@ public static class SynapseExtensions
 
     public static Team GetTeam(this RoleType role) => Server.Get.Host.ClassManager.Classes.SafeGet(role).team;
 
-    public static Room GetSynapseRoom(this RoomIdentifier identifier) => Map.Get.Rooms.FirstOrDefault(x => x.Identifier == identifier);
+    public static Room GetSynapseRoom(this RoomIdentifier identifier) => Map.Get.Rooms.FirstOrDefault(x => x?.Identifier == identifier);
 
-    public static Generator GetGenerator(this Scp079Generator generator079) => Map.Get.Generators.FirstOrDefault(x => x.GameObject == generator079.gameObject);
+    public static Generator GetGenerator(this Scp079Generator generator079) => Map.Get.Generators.FirstOrDefault(x => x?.GameObject == generator079.gameObject);
```
[the remark this answers is not in the document]
Comment (AlmightyLks) [reply Synapse/SynapseExtensions.cs:60]: Using Object.Destroy?

**Comment by warquys** [comment]
I created a [plugin to edit maps](https://github.com/VT-DevGiT/VT-MapEditor) and place schematics, but that only happened when I deleted schematics, doors, generators, etc. They happen that they are not removed from the list which causes null values ​​and therefore errors

File: Synapse/Api/Map.cs (hunk 1 above)
Comment (warquys) [reply Synapse/Api/Map.cs:132]: I don't understand

File: Synapse/SynapseExtensions.cs [hunk 4]
```diff
@@ -55,23 +55,24 @@ public static class SynapseExtensions
 
     public static Team GetTeam(this RoleType role) => Server.Get.Host.ClassManager.Classes.SafeGet(role).team;
 
-    public static Room GetSynapseRoom(this RoomIdentifier identifier) => Map.Get.Rooms.FirstOrDefault(x => x.Identifier == identifier);
+    public static Room GetSynapseRoom(this RoomIdentifier identifier) => Map.Get.Rooms.FirstOrDefault(x => x?.Identifier == identifier);
```
[the remark this answers is not in the document]
Comment (warquys) [reply Synapse/SynapseExtensions.cs]: yes i remov

File: Synapse/Api/Enum/Effect.cs [hunk 5]
```diff
@@ -157,5 +157,14 @@ public enum Effect
...
         Vitality,
+        /// <summary>
+        /// Increases movement speed and reload/draw speed for weapons and Stamina drain rate.
+        /// </summary>
+        /// <remarks>0 = Disabled, 1 = 1xScp1853, 2 = 2xScp1853...</remarks>
+        Scp1853,
+        /// <summary>
+        /// Do a blink
+        /// </summary>
+        Visuals173Blink,
```
Comment (Bonjemus) [inline Synapse/Api/Enum/Effect.cs:168]: I already added this effect.

Comment (warquys) [reply Synapse/Api/Enum/Effect.cs:168]: maby

File: Synapse/Api/Map.cs (hunk 1 above)
Comment (warquys) [reply Synapse/Api/Map.cs:132]: it was the default annoncement

**Pull request merged**